x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/components/create_analytics_form/create_analytics_form.tsx

```diff
@@ -186,8 +196,21 @@ export const CreateAnalyticsForm: FC<CreateAnalyticsFormProps> = ({ actions, sta
         setEstimatedModelMemoryLimit(expectedMemoryWithoutDisk);
       }
 
+      const fieldSelection: FieldSelectionItem[] = resp.field_selection;
+
+      let hasRequiredFields = false;
+      if (fieldSelection) {
```

Comment:
Is this `if` part necessary? Asking because the assignment looks like it will always be an array. Maybe this whole part can even become something like: 

```
const hasRequiredFields = resp.field_selection.some(field => field.is_included === true && field.is_required === false);
```

Reply:
Wrapping in an `if` statement is mostly an extra safety precaution for something going wrong silently in the backend or changes to it that would make `field_selection` `undefined`. The for loop prevents us from having to create a function on every go through. Though in this case the optimization is probably not significant. 🤔 

Your suggestion does offer better readability, though, so I'm happy to change it.

Reply:
Thanks for clarifying! Maybe we can then update the response type to something like `field_selection?: FieldSelectionItem[];` so TypeScript will force us to consider that the field could be not returned?

Reply:
Type updated in 44772641002802971036c772d68068171bf85dfa